src/fixml/modules/workflow/parse.py

```diff
@@ -99,50 +99,50 @@ def get_completeness_score(self, score_format: str = 'fraction', verbose: bool =
-
-            return md_repr
+    def as_markdown(self, add_quarto_header: bool = False) -> str:
 
         score = self.get_completeness_score(score_format='fraction')
         summary_df = self.evaluation_report[['ID', 'Title', 'is_Satisfied', 'n_files_tested']]
-        details = self.evaluation_report[['ID', 'Title', 'Requirement', 'Observations', 'Function References']].to_dict(orient='records')
-
-        export_content = dict()
-        export_content['Title'] = 'Test Evaluation Report'
-        export_content['Report Areas'] = []
-        export_content['Report Areas'].append({'Title': 'Summary', 'Completeness Score': score, 'Completeness Score per Checklist Item': '\n\n' + summary_df.to_markdown(index=False)})
-        export_content['Report Areas'].append({'Title': 'Details', 'Report Detail': details})
 
-        return _get_md_representation(export_content, 1)
+        response = self.response
+        call_results = response.call_results
+
+        metadata = {
+            "template_path": self.export_template.filename,
+        }
+        run_details = {
+            "checklist_path": response.checklist.path,
+            "repo_path": response.repository.path,
+            "head_commit": response.repository.git_commit,
+            "start_time": min([x.start_time for x in call_results]),
+            "end_time": max([x.end_time for x in call_results]),
+            "time_taken": max([x.end_time for x in call_results]) - min([x.start_time for x in call_results]),
+            "input_token_count": sum([x.tokens_used.input_count for x in call_results]),
+            "output_token_count": sum([x.tokens_used.output_count for x in call_results]),
+            "successful_count": sum([x.success for x in call_results]),
+            "failure_count": sum([not x.success for x in call_results]),
+            "success_perc": sum([x.success for x in call_results]) / len(call_results),
+            "files_evaluated": [file for x in call_results for file in x.files_evaluated],
+            "model_name_used": response.model.name,
+        }
+        eval_summary = {
+            "table": summary_df.to_markdown(index=False),
+            "score": score
+        }
+        eval_details = self.evaluation_report[['ID', 'Title', 'Requirement', 'Observations', 'Function References']].to_dict(orient='records')
+
+        vars = {
+            "quarto_header": add_quarto_header,
+            "title": "Test Evaluation Report",
+            "metadata": metadata,
+            "run_details": run_details,
+            "eval_summary": eval_summary,
+            "eval_details": eval_details
+        }
+        return self.export_template.render(**vars)
 
     def as_quarto_markdown(self) -> str:
-        header = '---\ntitle: "Test Evaluation Report"\nformat:\n  html:\n  code-fold: true\n---\n\n'
-        return header + self.as_markdown()
+        return self.as_markdown(add_quarto_header=True)
 
     def export_evaluation_report(self, output_path,
```

Comment:
I think `self.export` (i.e. ExportableMixin.export) provides a better handling of invalid output format?

```suggestion
    def export_evaluation_report(self, output_path, exist_ok: bool = False) -> None:
        self.export(output_path, exist_ok)
```

Reply:
Good find, this method is not referenced except for one place i.e. `RepositoryActions().evaluate()`. I would rather to remove this function entirely and make the usage to call `export()` directly